README.md

```diff
@@ -1,16 +1,817 @@
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * source_array[i] )
+    i += 1
+  end
+  return new
+end
+
+source_array = [1,2,3,4,5]
+p map_to_negativize(source_array)
+p map_to_no_change(source_array)
+p map_to_double(source_array)
+p map_to_square(source_array)
+```
+
+#===============================================================================
+
+# Writing Your Own Enumerables With Blocks
+
+## Learning Goals
+
+* Identify duplication that can be avoided with blocks
+* State two ways of constructing blocks
+* Execute a block from within a method
+* State the purpose of the `yield` keyword
+* Pass data between methods and blocks
+
+## Introduction
+
+As you saw in the previous lesson, there's a lot of repeated code between our
+various `map`-based methods. This should trigger an "allergic" reaction in you,
+you learned to DRY your code out. In this lesson we'll learn to DRY out our
+home-grown enumerable methods using a powerful language feature of Ruby: the
+_block_.
+
+## Identify Duplication That Can Be Avoided With Blocks
+
+Let's look at this solution code from the previous lesson:
+
+```ruby
+def map_to_negativize(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * -1 ) # <== Unique ork
+    i += 1
+  end
+  return new
+end
+
+def map_to_no_change(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+
+def map_to_double(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * 2 ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+
+def map_to_square(source_array)
+  new = []
+  i = 0
+  while i < source_array.length do
+    new.push( source_array[i] * source_array[i] ) # <== Unique work
+    i += 1
+  end
+  return new
+end
+```
+
+As you can see, the only thing that varies between these methods is the line
+which we've labeled with Unique work. When you learned to create methods, you
+probably learned that the way to _abstract_ the method is to pass in the stuff
+that varies as an _argument_ in the _call_ of the method. We'd like to do the
+same thing here.
+
+But in Ruby we know we can pass `String`s, `Array`s, `Hash`es, and `Float`s can
+we pass the abstraction of "work to do later?" With _blocks_ that's exactly
+what we do! We pass work, to be executed later, as an _implicit_ argument.
+
+## State Two Ways of Constructing Blocks
+
+We denote blocks in Ruby using two syntaxes:
+
+1. Curly brace `{}`
+2. `do...end`
+
+We use curly brace when the work inside is generally only one expression. Since
+expressions in Ruby _always_ return a value, the return value of the block will
+the value of that single expression.
+
+We use `do...end` when the work takes multiple lines. Just like methods,
+whatever is on the last line of a `do...end` block will be the return value of
+the block.
```

Comment:
is this true? does it matter? Just curious myself because I've mixed and matched them, using `{}` for multi line methods without issue. I just figured it was more of Ruby wanting to provide options similar to other programming languages

Reply:
It's a strong stylistic recomm https://github.com/airbnb/ruby#syntax